src/Splat.Microsoft.Extensions.DependencyInjection/MicrosoftDependencyResolver.cs

```diff
@@ -208,34 +205,31 @@ public virtual void UnregisterAll(Type? serviceType, string? contract = null)
 
         lock (_syncLock)
         {
-            switch (contract)
+            if (_serviceCollection is null)
```

Comment:
this is a behaviour change? the previous version did a null check on contract and threw on L228

"throw new ArgumentException("There must be a valid contract if there is no service collection.", nameof(contract));"

Reply:
this isn't an issue, the branch of code can't be hit. have put a note to come back to it later and look at cleanup